composer/algorithms/ema/ema.py

```diff
@@ -111,154 +116,172 @@ class EMA(Algorithm):
             )
     """
 
-    def __init__(self, half_life: str, update_interval: Optional[str] = None, train_with_ema_weights: bool = False):
-        self.half_life = half_life
-        self.update_interval = update_interval
-        self.train_with_ema_weights = train_with_ema_weights
-
+    def __init__(self,
+                 half_life: Optional[str] = None,
+                 smoothing: Optional[float] = None,
+                 ema_start: str = '0.0dur',
+                 update_interval: Optional[str] = None):
         self.ema_model = None
         self.training_model = None
+        self.ema_weights_active = False
+        self.ema_started = False
+        self.serialized_attributes = ['ema_model', 'training_model', 'ema_weights_active', 'ema_started']
+
+        # Verify that either half_life or smoothing has been specified
+        if half_life is None and smoothing is None:
+            raise ValueError(f'Either half_life or smoothing must be specified')
 
-        self.serialized_attributes = [
-            'ema_model',
-            'training_model',
-        ]
+        # Verify that only one of half_life or smoothing has been specified
+        if half_life is not None and smoothing is not None:
+            raise ValueError(f'Only one of  half_life or smoothing can be specified')
```

Comment:
I initially had things that way, but I think it's a little ambiguous. As in, ignoring what is/isn't a default, if a user specifies both `half_life` and `smoothing` it's ambiguous which they actually want. If one automatically gets overridden by the other, things might silently do something that the user doesn't want or expect. I could raise a warning for this, but I  don't know if it would be noticeable enough.

Reply:
My biggest concern is that both "half_life" and "smoothing" are labeled as optional in the docstring, but the user must specify one. I guess there isn't anyway around this?

Reply:
I could make `half_life` not optional, and then have `smoothing` override it, but that feels error prone to me. Which do you prefer?

Reply:
Maybe it's better to fail from no inputs than to silently do something with both inputs. Feel free to ignore haha

Reply:
Ok, I added a default value for `half_life`. We'll accept the annoyance if someone does want to run with `smoothing` and has to remember to turn off `half_life` since presumably that means they know what they're about when it comes to EMA.